objc/core/A0LockReact.m

```diff
@@ -44,6 +49,19 @@ - (void)configureLockWithClientId:(NSString *)clientId domain:(NSString *)domain
     _lock = [A0Lock newLockWithClientId:clientId domain:domain];
 }
 
+- (void)hideWithCallback:(A0LockCallback)callback {
+    if (self.shown) {
```

Comment:
@hzalaz that scenario was what I was trying to avoid. Wanted the call to be safe to attach to some state change event which may not be dependent upon the lock being shown.

e.g. we have a listener on a firebase variable that detects when an app version value has changed. If the version requires an updated we ensure the login screen is hidden by calling lock.hide() without having to keep track of whether or not it's shown.